out/can_handlers.c

```diff
@@ -18,23 +19,27 @@
         f(dst, src, length); \
     } \
 
-IMPL_HANDLER(can_database_vcu_internal_states_t, can_database_vcu_internal_states_unpack)
-IMPL_HANDLER(can_database_vcu_driver_inputs_t, can_database_vcu_driver_inputs_unpack)
-IMPL_HANDLER(can_database_pm100_fault_codes_t, can_database_pm100_fault_codes_unpack)
-IMPL_HANDLER(can_database_pm100_internal_states_t, can_database_pm100_internal_states_unpack)
-IMPL_HANDLER(can_database_pm100_current_info_t, can_database_pm100_current_info_unpack)
-IMPL_HANDLER(can_database_pm100_motor_position_info_t, can_database_pm100_motor_position_info_unpack)
+IMPL_HANDLER(can_database_can_c_vcu_internal_states_t, can_database_can_c_vcu_internal_states_unpack)
+IMPL_HANDLER(can_database_can_c_vcu_driver_inputs_t, can_database_can_c_vcu_driver_inputs_unpack)
+IMPL_HANDLER(can_database_can_c_pm100_fault_codes_t, can_database_can_c_pm100_fault_codes_unpack)
+IMPL_HANDLER(can_database_can_c_pm100_internal_states_t, can_database_can_c_pm100_internal_states_unpack)
+IMPL_HANDLER(can_database_can_c_pm100_current_info_t, can_database_can_c_pm100_current_info_unpack)
+IMPL_HANDLER(can_database_can_c_pm100_motor_position_info_t, can_database_can_c_pm100_motor_position_info_unpack)
+IMPL_HANDLER(can_database_can_s_oct_gps_2_t, can_database_can_s_oct_gps_2_unpack)
+IMPL_HANDLER(can_database_can_s_oct_gps_1_t, can_database_can_s_oct_gps_1_unpack)
 
 /**
  * @brief   Table of CAN message IDs and associated unpacking functions
  */
 static const can_handler_t can_handler_table[] = {
```

Comment:
I don't see a way how the OCT software will be able to know which handler to use if there are two messages with the same CAN ID on the CAN-C and CAN-S buses? The only way they are identified is by `CAN_C` or `CAN_S` within the frame IDs. 

I suggest we split it into two handler tables, one for CAN-C and one for CAN-S. Then in the OCT software, have two instances of the CAN unpacking service. Each service instance then gets a pointer to the handlers table which it should use and a pointer to an RTCAN handle for the corresponding bus.